src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,74 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const pageNumbers = getNumbers(1, totalPages);
+
+  const handlePageChange = (page: number) => {
+    if (page !== currentPage && page > 0 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn('page-item', { disabled: currentPage === 1 })}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => handlePageChange(currentPage - 1)}
+          >
+            «
+          </a>
+        </li>
+        {pageNumbers.map(pageNumber => (
+          <li
+            key={pageNumber}
+            className={cn('page-item', {
+              active: currentPage === pageNumber,
+            })}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href={`#${pageNumber}`}
+              onClick={() => handlePageChange(pageNumber)}
+            >
+              {pageNumber}
+            </a>
+          </li>
+        ))}
+        <li
+          className={cn('page-item', { disabled: currentPage === totalPages })}
+        >
+          <a
+            data-cy="nextLink"
+            className="page-link"
+            href="#next"
+            aria-disabled={currentPage === totalPages}
+            onClick={() => handlePageChange(currentPage + 1)}
```

Comment:
To improve accessibility, consider adding an 'onKeyDown' event here as well, to enable keyboard navigation.